### PR TITLE
8267875: Shenandoah: Duplicated code in ShenandoahBarrierSetC2::ideal_node()

### PR DESCRIPTION
**Pull request by zhengyu123**
Node::set_req_X definition as:
void Node::set_req_X(uint i, Node *n, PhaseGVN *gvn) {
  PhaseIterGVN* igvn = gvn->is_IterGVN();
  if (igvn == NULL) {
    set_req(i, n);
    return;
  }
  set_req_X(i, n, igvn);
}

ShenandoahBarrierSetC2::ideal_node() duplicates above logic.

Test:
  hotspot_gc_shenandoah

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8267875](https://bugs.openjdk.java.net/browse/JDK-8267875): Shenandoah: Duplicated code in ShenandoahBarrierSetC2::ideal_node()

### Reviewers
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - **Reviewer**)
 * [Roland Westrelin](https://openjdk.java.net/census#roland) (@rwestrel - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4229/head:pull/4229` \
`$ git checkout pull/4229`

Update a local copy of the PR: \
`$ git checkout pull/4229` \
`$ git pull https://git.openjdk.java.net/jdk pull/4229/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4229`

View PR using the GUI difftool: \
`$ git pr show -t 4229`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4229.diff">https://git.openjdk.java.net/jdk/pull/4229.diff</a>

</details>

### CODE REVIEW

**Review by rkennke — APPROVED** [review]
LGTM. Please wait for @rwestrel approval.

**Review by rwestrel — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**
